extendr-api/tests/from_iterator_tests.rs

```diff
@@ -0,0 +1,59 @@
+//! Test cases for this were made by @JosiahParry.
+//!
+//!
+use extendr_api::prelude::*;
+
+// #[extendr]
+fn protect_lim2(n: i32) -> List {
+    #[derive(Debug)]
+    struct PlzBreak(i32);
+
+    let n = n as usize;
+
+    (0..n)
+        .into_iter()
+        .map(|xi| ExternalPtr::new(PlzBreak(xi as i32)))
+        .collect::<List>()
+}
+
+// #[extendr]
+fn prot_strs(n: i32) -> Strings {
+    let n = n as usize;
+    (0..n)
+        .into_iter()
+        .map(|_| Rstr::from_string("val"))
+        .collect::<Strings>()
+}
+
+#[test]
+fn test_from_iterator_collection() {
+    test!(
+        let s = prot_strs(10_000 * 7);
+        println!("Hello: {}", s[54]);
+        protect_lim2(10_000 * 7);
+    )
+}
+
+#[test]
+fn test_with_gc_torture() {
+    test!(
+        let x = vec![1, 4, 5, 6];
+        R!("gctorture(on = TRUE)")?;
+        let list: List = x.into_iter().collect();
+        R!("gctorture(on = FALSE)")?;
+        assert_eq!(list, list!(1, 4, 5, 6));
+    );
+}
```

Comment:
One improvement I would like here. I've noticed that I don't encounter segfaults unless the vector allocations are large (https://stat.ethz.ch/pipermail/r-devel/2023-April/082589.html). Could you try making a list of large vectors (n>100) instead of just scalars?

Reply:
Good idea. I'll add that later.